python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -290,146 +292,157 @@ def _makeBoxRegion(BBox, wcs, BBoxPadding):
-        the datastore and return a single catalog containing all reference objects
+        The spherical box region and filtering function are passed to the
+        generic loadRegion method which loads and filters the reference objects
+        from the datastore and returns a single catalog containing the filtered
+        set of reference objects.
 
         Parameters
         ----------
         bbox : `lsst.geom.box2I`
-            Box which bounds a region in pixel space
+            Box which bounds a region in pixel space.
         wcs : `lsst.afw.geom.SkyWcs`
-            Wcs object defining the pixel to sky (and inverse) transform for the space
-            of pixels of the supplied bbox
-        filterName : `str`
-            Name of camera filter, or None or blank for the default filter
-        epoch : `astropy.time.Time` (optional)
-            Epoch to which to correct proper motion and parallax,
-            or None to not apply such corrections.
-        photoCalib : None
-            Deprecated and ignored, only included for api compatibility
-        bboxPadding : `int`
-            Number describing how much to pad the input bbox by (in pixels), defaults
-            to 100. This parameter is necessary because optical distortions in telescopes
-            can cause a rectangular pixel grid to map into a non "rectangular" spherical
-            region in sky coordinates. This padding is used to create a spherical
-            "rectangle", which will for sure enclose the input box.  This padding is only
-            used to determine if the reference catalog for a sky patch will be loaded from
-            the data store, this function will filter out objects which lie within the
-            padded region but fall outside the input bounding box region.
+            Wcs object defining the pixel to sky (and inverse) transform for
+            the supplied ``bbox``.
+        filterName : `str` or `None`, optional
+            Name of camera filter, or `None` or blank for the default filter.
+        epoch : `astropy.time.Time` or `None`, optional
+            Epoch to which to correct proper motion and parallax, or `None`
+            to not apply such corrections.
+        photoCalib : `None`
+            Deprecated and ignored, only included for api compatibility.
 
         Returns
         -------
         referenceCatalog : `lsst.afw.table.SimpleCatalog`
-            Catalog containing reference objects inside the specified bounding box
+            Catalog containing reference objects inside the specified bounding
+            box (padded by self.config.pixelMargin).
 
         Raises
         ------
-        `lsst.pex.exception.RuntimeError`
-            Raised if no reference catalogs could be found for the specified region
-
-        `lsst.pex.exception.TypeError`
-            Raised if the loaded reference catalogs do not have matching schemas
+        RuntimeError
```

Comment:
The type of these Error was correct before the change

Reply:
We should not be raising pex_exception RuntimeError or TypeError in python code. They should only be raised in C++ code. It would be better to change this code to use the normal python exceptions. (I still don't understand why pybind11 doesn't translate those exceptions to the standard Python ones at the interface -- you can catch a pexExcept.RuntimeError as a normal RuntimeError so there's generally no reason to use the pex_exceptions code in python at all).

PS pex_exceptions is imported twice at the top of this file with different aliases.

Reply:
Dev guide: https://developer.lsst.io/python/style.html?highlight=pex_exceptions#python-exceptions-should-be-raised-and-checked-in-python-code

Reply:
I have no preference here. I think I just copy and pasted them from some code when I hacked this together quickly ages ago. I am happy for them to change, I just wanted to point out that the type info was not actually correct.